code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -406,17 +472,66 @@ func (sc *StorageSmartContract) verifyChallenge(t *transaction.Transaction,
 		prev = last.Created
 	}
 
-	var (
-		threshold = len(challReq.Validators) / 2
-		pass      = success > threshold ||
-			(success > failure && success+failure < threshold)
-		cct   = toSeconds(details.Terms.ChallengeCompletionTime)
-		fresh = challReq.Created+cct >= t.CreationDate
-	)
+	pass, fresh, threshold := challReq.isChallengePassed(
+		success, failure, details.Terms.ChallengeCompletionTime, t.CreationDate)
 
 	// verification, or partial verification
 	if pass && fresh {
 
+		blobber, err := sc.getBlobber(t.ClientID, balances)
+		if err != nil {
+			return "", common.NewError("verify_challenge",
+				"can't get blobber"+err.Error())
+		}
+
+		ogList, err := getOngoingPassedBlobbersList(balances)
```

Comment:
Access each MPT key only once. You called `getOngoingPassedBlobbersList` above.

Reply:
the reason I'm calling again is because the older one has been migrated and deleted in .Migrate()
this will fetch the latest partition

Reply:
If it has been deleted then you already know that. Use a local variable. 

MPT access takes hundreds of microseconds not nanoseconds. There should be no reason to read or save the same key more than once.